lib/kube/proxy/testing/kube_server/pods.go

```diff
@@ -143,15 +146,60 @@ func (s *KubeMockServer) DeletedPods(reqID string) []string {
 
 // parseDeleteCollectionBody parses the request body targeted to pod collection
 // endpoints.
-func parseDeleteCollectionBody(r io.Reader) (metav1.DeleteOptions, error) {
+func parseDeleteCollectionBody(r *http.Request) (metav1.DeleteOptions, error) {
 	into := metav1.DeleteOptions{}
-	data, err := io.ReadAll(r)
+	data, err := io.ReadAll(r.Body)
```

Comment:
nit: Close the body after draining.

```suggestion
	data, err := io.ReadAll(r.Body)
	_ = r.Body.Close()
```

Logging the Close error instead of swallowing is a good option too.